flocker/node/agents/blockdevice.py

```diff
@@ -1468,6 +1540,94 @@ def _get_system_mounts(self, volumes, compute_instance_id):
             if FilePath(partition.device) in device_to_dataset_id
         }
 
+    def _upgrade_1_0_0_filesystem(self, api, volumes):
+        """
+        Upgrade any old, Flocker 1.0.0-created filesystems.
+
+        Examine each volume attached to this host.  Determine if the filesystem
+        on the volume has the corresponding dataset_id recorded in its UUID.
+        This is how filesystems are created post-1.0.0 (see FLOC-2369).  If the
+        filesystem does not have this information recorded and the filesystem
+        is not mounted, put it there.  If the filesystem is mounted we cannot
+        modify it.  We will fix it when we notice it is unmounted.
+
+        The intent is for this to cause Flocker 1.0.0-created volumes to look
+        just like they would have if they had been created by Flocker >1.0.0.
+
+        :param IBlockDeviceAPI api: The storage driver responsible for the
+            volumes.
+        :param list volumes: The ``BlockDeviceVolumes`` that exist and will be
+            considered for upgrade.
+        """
+        mounted = {
+            FilePath(partition.device)
+            for partition
+            in psutil.disk_partitions()
+        }
+
+        compute_instance_id = api.compute_instance_id()
+        for volume in _attached_volumes(compute_instance_id, volumes):
+            try:
+                get_device_for_dataset_id(volume.dataset_id)
+                # If we found the dataset_id like this the filesystem is
+                # sufficiently new to not need any changes.  Proceed to the
+                # next volume.
+                continue
+            except KeyError:
+                # There is no filesystem on the volume because volume setup
```

Comment:
I think the order here is wrong. By doing this first it means that if you have missing filesystem you will always detach and attach, which is sloooooooow. So missing filesystem goes from being annoyance to being pretty terrible.

Instead we should first try get_device_path maybe? THough even then you get slowdowns if you restart dataset agents.... So still a problem.

Big picture point: upgrades shouldn't cause slow downs ever when not upgrading,and right now they do. Can we somehow disable upgrades once it's clear they're not necessary?